### ISSUE TITLE
helmfile repos don't sync repositories

### ISSUE THREAD

**Issue by dungdm93** [roboll/helmfile#1330]
`helmfile` don't sync repos in `helmfile.yaml`, from v0.115.0 to v0.119.1.
`helmfile` v0.114.0 works.

```bash
$ cat helmfile.yaml 
repositories:
- name: stable
  url:  https://kubernetes-charts.storage.googleapis.com
- name: bitnami
  url:  https://charts.bitnami.com/bitnami
- name: jetstack
  url:  https://charts.jetstack.io

helmfiles:
- path: cert-manager/helmfile.yaml

$ helm version
version.BuildInfo{Version:"v3.2.4", GitCommit:"0ad800ef43d3b826f31a5ad8dfbb4fe05d143688", GitTreeState:"clean", GoVersion:"go1.13.12"}

$ helm repo list
Error: no repositories to show

$ helmfile --version
helmfile version v0.119.1

$ helmfile repos

$ helm repo list
Error: no repositories to show
```

**Comment by dpiddockcmp** [comment]
I've just hit this same issue. 

It seems to happen when repositories are listed in YAML files on their own. If they have a `releases` then they get updated.

From a `helmfile build`:
```yaml
---
#  Source: 00-defaults.yaml

filepath: 00-defaults.yaml
helmBinary: helm
repositories:
- name: stable
  url: https://kubernetes-charts.storage.googleapis.com
templates: {}
missingFileHandler: ""
---
#  Source: 01-cert-manager.yaml

filepath: 01-cert-manager.yaml
helmBinary: helm
repositories:
- name: jetstack
  url: https://charts.jetstack.io
releases:
- chart: jetstack/cert-manager
    # etc
templates: {}
missingFileHandler: ""
```
`helmfile repos` adds and updates the `jetstack` repo but not `stable`. If we move `stable` to be in a file with a `releases` it also gets added correctly.

**Comment by mumoshu** [comment]
@dungdm93 @dpiddockcmp Thanks for reporting! I think this is a regression since https://github.com/roboll/helmfile/pull/1233 (cc/ @guillomep)

I'll fix it as part of #1383
